src/daft-schema/src/python/datatype.rs

```diff
@@ -355,61 +360,279 @@ impl PyDataType {
+        self.dtype.is_fixed_size_list()
+    }
+
+    pub fn is_list(&self) -> bool {
+        self.dtype.is_list()
+    }
+
+    pub fn is_struct(&self) -> bool {
+        self.dtype.is_struct()
+    }
+
+    pub fn is_map(&self) -> bool {
+        self.dtype.is_map()
+    }
+
+    pub fn is_extension(&self) -> bool {
+        self.dtype.is_extension()
+    }
+
+    pub fn is_image(&self) -> bool {
+        self.dtype.is_image()
+    }
+
+    pub fn is_fixed_shape_image(&self) -> bool {
+        self.dtype.is_fixed_shape_image()
+    }
+
+    pub fn is_embedding(&self) -> bool {
+        self.dtype.is_embedding()
+    }
+
+    pub fn is_tensor(&self) -> bool {
+        self.dtype.is_tensor()
+    }
+
+    pub fn is_fixed_shape_tensor(&self) -> bool {
+        self.dtype.is_fixed_shape_tensor()
+    }
+
+    pub fn is_sparse_tensor(&self) -> bool {
+        self.dtype.is_sparse_tensor()
+    }
+
+    pub fn is_fixed_shape_sparse_tensor(&self) -> bool {
+        self.dtype.is_fixed_shape_sparse_tensor()
+    }
+
+    pub fn is_python(&self) -> bool {
+        self.dtype.is_python()
+    }
+
+    pub fn is_numeric(&self) -> bool {
+        self.dtype.is_numeric()
+    }
+
+    pub fn is_integer(&self) -> bool {
+        self.dtype.is_integer()
+    }
+
+    pub fn is_logical(&self) -> bool {
+        self.dtype.is_logical()
+    }
+
+    pub fn is_temporal(&self) -> bool {
+        self.dtype.is_temporal()
+    }
+
+    pub fn fixed_size(&self) -> PyResult<usize> {
+        self.dtype.fixed_size().ok_or_else(|| {
+            PyAttributeError::new_err(format!(
+                "The data type {} does not contain a fixed size",
+                self.dtype
+            ))
+        })
+    }
+
+    pub fn fixed_shape(&self) -> PyResult<Vec<u64>> {
+        self.dtype.fixed_shape().ok_or_else(|| {
+            PyAttributeError::new_err(format!(
+                "The data type {} does not contain a fixed shape",
+                self.dtype
+            ))
+        })
+    }
+
+    pub fn time_unit(&self) -> PyResult<PyTimeUnit> {
+        self.dtype.time_unit().map(|tu| tu.into()).ok_or_else(|| {
+            PyAttributeError::new_err(format!(
+                "The data type {} does not contain a time unit",
+                self.dtype
+            ))
+        })
+    }
+
+    pub fn time_zone(&self) -> PyResult<Option<String>> {
+        self.dtype
+            .time_zone()
+            .map(|tz| tz.map(|tz| tz.to_string()))
+            .ok_or_else(|| {
+                PyAttributeError::new_err(format!(
+                    "The data type {} does not contain a timezone",
+                    self.dtype
+                ))
+            })
+    }
+
+    pub fn image_mode(&self) -> PyResult<ImageMode> {
+        self.dtype.image_mode().ok_or_else(|| {
+            PyAttributeError::new_err(format!(
+                "The data type {} does not contain an image mode",
+                self.dtype
+            ))
+        })
```

Comment:
Ok, here I think the problem may be surfaced. If the image_mode is None, we are erroring rather than returning None. The attribute is valid, but the value is None. It's interesting how the Image type's mode is optional while the FixedShapeImage mode is not.